Quadriga/src/main/java/edu/asu/spring/quadriga/rest/SearchNetworkRestController.java

```diff
@@ -108,10 +113,19 @@
             return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
         }
 
+        // Fetch ConceptPower entries related to the conceptId
+        ConceptpowerReply reply = conceptpowerConnector.getById(conceptId);
+        List<String> alternativeIdsForConcept = null;
+        if (reply != null && reply.getConceptEntry().size() > 0) {
+            alternativeIdsForConcept = reply.getConceptEntry().get(0).getAlternativeIdList();
+        }
+        if(alternativeIdsForConcept == null){
+            alternativeIdsForConcept = new ArrayList<String>();
+        }  
```

Comment:
service layer is responsible for business logic, and that the alternative ids need to be used rather than should the main id is business logic. If you require every caller of the method to get the required ids from conceptpower, you a) duplicate code and b) put business logic into layers that shouldn't contain any. If the logic for some reason changes later, e.g. we'll have an "alternative ids" field and a "representing ids" field and we'll need to use "representing ids" rather than "alternative ids", then you would have to change every single caller of the business logic method to retrieve the correct ids, rather than once in the business logic. Make sense?